src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/X509Pal.CustomExtensions.cs

```diff
@@ -50,9 +50,21 @@ public void DecodeX509KeyUsageExtension(byte[] encoded, out X509KeyUsageFlags ke
                         Debug.Assert(cbDecoded >= sizeof(CRYPT_BIT_BLOB));
                         CRYPT_BIT_BLOB* pBlob = (CRYPT_BIT_BLOB*)pvDecoded;
                         keyUsagesAsUint = 0;
-                        if (pBlob->pbData != null)
+                        byte* pbData = pBlob->pbData;
+
+                        if (pbData != null)
                         {
-                            keyUsagesAsUint = *(uint*)(pBlob->pbData);
+                            Debug.Assert((uint)pBlob->cbData < 3, "Unexpected length for X509_KEY_USAGE data");
+
+                            switch (pBlob->cbData)
+                            {
+                                case 1:
+                                    keyUsagesAsUint = *pbData;
+                                    break;
+                                case 2:
+                                    keyUsagesAsUint = *(ushort*)(pbData);
+                                    break;
```

Comment:
We have tests appropriately covering all of the various cases here?

Reply:
Yeah, there are only 9 bits defined in the vector, so it only gets a second byte when DecipherOnly is set.  So 99% of the time cbData is 1, and we have a test for the 0, and a couple tests for the 2.